### PR TITLE
GCP Libraries BOM version to 26.14.0

### PR DESCRIPTION
**Pull request by mutianf**
Update google_cloud_platform_libraries_bom version. Update gax, spanner and grpc versions to be consistent with the versions in google_cloud_platform_libraries_bom. 

------------------------

Thank you for your contribution! Follow this checklist to help us incorporate your contribution quickly and easily:

 - [ ] Mention the appropriate issue in your description (for example: `addresses #123`), if applicable. This will automatically add a link to the pull request in the issue. If you would like the issue to automatically close on merging the pull request, comment `fixes #<ISSUE NUMBER>` instead.
 - [ ] Update `CHANGES.md` with noteworthy changes.
 - [ ] If this contribution is large, please file an Apache [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

See the [Contributor Guide](https://beam.apache.org/contribute) for more tips on [how to make review process smoother](https://beam.apache.org/contribute/get-started-contributing/#make-the-reviewers-job-easier).

To check the build health, please visit [https://github.com/apache/beam/blob/master/.test-infra/BUILD_STATUS.md](https://github.com/apache/beam/blob/master/.test-infra/BUILD_STATUS.md)

GitHub Actions Tests Status (on master branch)
------------------------------------------------------------------------------------------------
[Build python source distribution and wheels](https://github.com/apache/beam/actions?query=workflow%3A%22Build+python+source+distribution+and+wheels%22+branch%3Amaster+event%3Aschedule)
[Python tests](https://github.com/apache/beam/actions?query=workflow%3A%22Python+Tests%22+branch%3Amaster+event%3Aschedule)
[Java tests](https://github.com/apache/beam/actions?query=workflow%3A%22Java+Tests%22+branch%3Amaster+event%3Aschedule)
[Go tests](https://github.com/apache/beam/actions?query=workflow%3A%22Go+tests%22+branch%3Amaster+event%3Aschedule)

See [CI.md](https://github.com/apache/beam/blob/master/CI.md) for more information about GitHub Actions CI.

### CODE REVIEW

**Comment by suztomo** [comment]
Looks good to me. Let's see the test results.

**Comment by Abacn** [comment]
Run Java_IOs_Direct PreCommit

**Comment by Abacn** [comment]
Run SQL PreCommit

**Comment by Abacn** [comment]
beam_PreCommit_Java_Commit testParDoRequiresStableInputPortable  known flake: #21333
beam_PreCommit_Java_PVR_Flink_Batch_Commit known timeout flake: #26301

**Comment by suztomo** [comment]
Run SQL Postcommit

**Comment by suztomo** [comment]
Run SQL Postcommit

**Comment by suztomo** [comment]
@Abacn Do we wait until the flakes are resolved or merge this without waiting for them?

**Comment by Abacn** [comment]
> @Abacn Do we wait until the flakes are resolved or merge this without waiting for them?

The traceback shows a grpc exception and the version bump included grpc, we can wait until the trigger run ends. Jenkins was in restart mode and there is a long queue and now is back. Trigger job should be completed soon.

**Comment by suztomo** [comment]
Run Java PreCommit

**Comment by mutianf** [comment]
Do we have to update the grpc version to be in sync with libraries bom? If grpc is the only concern / causing failures I could revert the grpc change

File: buildSrc/src/main/groovy/org/apache/beam/gradle/BeamModulePlugin.groovy
Comment (suztomo) [inline buildSrc/src/main/groovy/org/apache/beam/gradle/BeamModulePlugin.groovy:539]: Memo. 1.54.0 is correct value.

> gRPC: 1.54.0

https://github.com/googleapis/java-cloud-bom/releases/tag/v26.14.0

**Review by Abacn — APPROVED** [review]
Test passed, should be good to go

**Pull request merged**
